pkg/distributor/distributor.go

```diff
@@ -92,7 +96,9 @@ type Distributor struct {
 	ingestersRing    ring.ReadRing
 	validator        *Validator
 	pool             *ring_client.Pool
-	streamSharder    StreamSharder
+
+	rateStore RateStore
+	sharder   func(stream logproto.Stream, desiredRate int, rateStore RateStore) (int, error)
```

Comment:
Would it make more sense to have the `rateStore` return a known rate and just make sure we get the right number of shards out of `shardStream`? That would be a bit simpler and cover stream sharding in one logical test

Reply:
hmmm but isn't `rateStore` returning a known rate for a given stream? or do you mean like moving from an interface to a func?

>  just make sure we get the right number of shards out of shardStream

the `shardStream` is already tested (implemented by an older PR). So what I only felt the necessity to add tests for the new `calculateShards`, as it relies on things that weren't existent when the `streamStream` tests were written.
I'm planning in adding tests for `shardCountFor` though, WDYT?